kiwi/bootloader/config/systemd_boot.py

```diff
@@ -0,0 +1,159 @@
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import os
+import re
+import glob
+from string import Template
+from typing import Dict
+
+# project
+from kiwi.path import Path
+from kiwi.bootloader.template.systemd_boot import BootLoaderTemplateSystemdBoot
+from kiwi.bootloader.config.bootloader_spec_base import BootLoaderSpecBase
+from kiwi.system.kernel import Kernel
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiTemplateError,
+    KiwiBootLoaderTargetError,
+    KiwiKernelLookupError
+)
+
+
+class BootLoaderSystemdBoot(BootLoaderSpecBase):
+    def create_loader_image(self, target: str) -> None:
+        """
+        Handle EFI images
+
+        For systemd boot EFI images are provided along with systemd.
+        Thus the only action taken care of is the creation of the
+        ESP path
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+        """
+        self.efi_boot_path = self.create_efi_path()
+
+    def setup_loader(self, target: str) -> None:
+        """
+        Setup ESP for systemd-boot using bootctl and kernel-install
+
+        For disk images only, setup the ESP layout for systemd boot
+        using bootctl. All data will be loaded from the ESP. The
+        kernel-install script provided with systemd is used to
+        manage kernel updates and the respective loader entries.
+        Distributions integrating well with systemd runs kernel-install
+        as part of their kernel packaging. In kiwi we use kernel-install
+        for the initial boot setup
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+            Currently only the disk identifier is supported
+        """
+        if target != self.target.disk:
+            raise KiwiBootLoaderTargetError(
+                'systemd-boot is only supported with the EFI disk image target'
+            )
+        boot_options = self.custom_args['boot_options']
+        self._mount_system(
+            boot_options.get('root_device'),
+            boot_options.get('boot_device'),
+            boot_options.get('efi_device'),
+            boot_options.get('system_volumes')
+        )
+        kernel_info = Kernel(self.root_mount.mountpoint).get_kernel()
+        if not kernel_info:
+            raise KiwiKernelLookupError(
+                'No kernel in image tree found'
+            )
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint, 'bootctl', 'install',
+                '--esp-path=/boot/efi',
+                '--no-variables',
+                '--entry-token', 'os-id'
```

Comment:
This should probably be something more descriptive than `os-id` no? Ideally the entry-token name is configurable by the user.

Reply:
according to the documentation ```entry-token os-id``` takes the information from ```/etc/os-release``` which I think is a nice idea. In my tests with Fedora this was also the case and the naming looked good to me. Also as ```/etc/os-release``` is distro specific provided data it felt like a good concept

Reply:
Ahh, I forgot we added something like that, makes sense

Reply:
:)